src/app/shared/components/response-answers/response-answers.component.ts

```diff
@@ -7,64 +9,86 @@ import { Component, OnInit, Input, ChangeDetectorRef } from '@angular/core';
-          } else  if (responseQuestion.questionType === 'checkbox') {
-            answer =  this.getCheckboxAnswer(responseQuestion, item);
-          } else {
-            if (item.is_other) {
-              answer = {
-                id: responseQuestion.id,
-                questionType: responseQuestion.questionType,
-                questionText: responseQuestion.questionText,
-                answer: item.other_option_answer
-              };
+    if (!isObjectEmpty(this.response)) {
+      this.answers = this.response.answers;
+      if (this.answers && this.answers.length) {
+        this.questions = this.getQuestions();
+        const responseAnswers = [];
+        this.answers.map((item) => {
+          let answer = {};
+          if (this.questions && this.questions.length) {
+            const responseQuestion = this.questions.find((question) => +question.id === +item.question_id);
+            if (responseQuestion) {
+            if (responseQuestion.questionType === 'multiple_choice' || responseQuestion.questionType === 'dropdown') {
+              answer = item.is_other ? this.getMultiChoiceAnswer(responseQuestion, item) :
+               this.getMultiChoiceAnswer(responseQuestion, item.answer);
+            } else  if (responseQuestion.questionType === 'checkbox') {
+              answer =  this.getCheckboxAnswer(responseQuestion, item);
             } else {
-              answer = {
-                id: responseQuestion.id,
-                questionType: responseQuestion.questionType,
-                questionText: responseQuestion.questionText,
-                answer: item.answer
-              };
+              if (item.is_other) {
+                answer = {
+                  id: responseQuestion.id,
+                  questionType: responseQuestion.questionType,
+                  questionText: responseQuestion.questionText,
+                  answer: item.other_option_answer
+                };
+              } else {
+                answer = {
+                  id: responseQuestion.id,
+                  questionType: responseQuestion.questionType,
+                  questionText: responseQuestion.questionText,
+                  answer: item.answer
+                };
+              }
             }
-
           }
-        }
-        }
-        if (this.showOnlyLongTextAnswer) {
-          if (answer['questionType'] === 'long_text') {
+          }
+          if (this.showOnlyLongTextAnswer) {
+            if (answer['questionType'] === 'long_text') {
+              responseAnswers.push(answer);
+            }
+          } else {
             responseAnswers.push(answer);
           }
-        } else {
-          responseAnswers.push(answer);
-        }
-      });
-      return responseAnswers;
+        });
+        return responseAnswers;
+      }
     }
     return;
   }
 
+  getQuestions() {
+    let questions;
+    if (this.showOnlyLongTextAnswer) {
```

Comment:
@Nishanthh Here both the if and the else blocks have the same code running. Was something missed?

Reply:
@avikt No Nothing missed here, its a old flow when no tabs in public responses, now modified